### PR TITLE
rm: security: add page for tpm2 hsm support

### PR DESCRIPTION
**Pull request by ricardosalveti**
## Readiness

* [ ] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

Instructions for using TPM2 devices with LmP.

## Checklist

_Optional. Add a 'x' to steps taken._
_You can fill this out after opening the PR. "Did I..."_

* [ ] Run spelling and grammar check, preferably with linter.
* [ ] Avoid changing any header associated with a link/reference.
* [x] Step through instructions (or ask someone to do so).
* [ ] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [ ] Match tone and style of page/section.
* [ ] Run `make linkcheck`.
* [x] View HTML in a browser to check rendering.
* [ ] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [ ] follow best practices for commits.
  * [ ] Descriptive title written in the imperative.
  * [ ] Include brief overview of QA steps taken.
  * [ ] Mention any related issues numbers.
  * [ ] End message with sign off/DCO line (`-s, --signoff`).
  * [ ] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [x] Request PR review by a technical writer and at least one peer.

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 59835de03ab564a0638e6d848de7b429f273d520 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1992/docs/artifacts/html/index.html

File: source/reference-manual/security/secure-elements/secure-element.tpm.rst [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+and by the configuration manager (``fioconfig``).
+
+TPM 2 Software Stack
+--------------------
+
+The `TPM2 software stack`_ is a complex set of software components that enable the integration and use
+of the TPM2 device. The software stack is comprised of multiple layers and abstractions, each providing
+different functionalities and interfaces to interact with the TPM2 chip.
+
+The main software component used is `TPM2 TSS (TPM Software Stack)`_. This layer provides a software framework
```
Comment (kprosise) [inline source/reference-manual/security/secure-elements/secure-element.tpm.rst:24]:
```suggestion
The main software component used is `TPM 2 Software Stack (TPM2 TSS)`_. This layer provides a software framework
```
Changing reference at bottom to match as well

File: source/reference-manual/security/secure-elements/secure-element.tpm.rst [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+To register with support for TPM 2 PKCS#11:
+
+.. code-block:: shell
+
+    lmp-device-register -n <device-name> -m /usr/lib/pkcs11/libtpm2_pkcs11.so -S <so-pin> -P <user-pin>
+
+.. _TPM2 software stack:
+   https://tpm2-software.github.io/
+
+.. _TPM2 TSS (TPM Software Stack):
```
Comment (kprosise) [inline source/reference-manual/security/secure-elements/secure-element.tpm.rst:97]:
```suggestion
.. _TPM 2 Software Stack (TPM2 TSS):
```

**Review by kprosise — COMMENTED** [review]
First pass, mostly addressing linter findings. I'll do another review after we get the linter to mostly be happy.

File: source/reference-manual/security/secure-elements/secure-element.tpm.rst [hunk 3]
```diff
@@ -0,0 +1,119 @@
+.. highlight:: sh
+
+.. _ref-secure-element.tpm:
+
+Trusted Platform Module (TPM 2.0)
```
[the remark this answers is not in the document]
Comment (ricardosalveti) [reply source/reference-manual/security/secure-elements/secure-element.tpm.rst:5]: Here I wanted to state that the version we support is 2.0 (kind of how all the libraries are named as well).

[the remark this answers is not in the document]
Comment (ricardosalveti) [reply source/reference-manual/security/secure-elements/secure-element.tpm.rst:5]: Just saw the note, nice.

**Comment by doanac** [comment]
Docs for adc2b3b3a3660ef16b39cac15f435f4ad47b03cd are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1993/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for c0b65a61d92b22b835996f95089f89eacb105d24 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1994/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 351981787af76db7989c58c53eb4f270affdb5e3 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1995/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 520e702e3380ad6d1081d57c4291baa3bae61fec are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1996/docs/artifacts/html/index.html

**Comment by kprosise** [comment]
Working on the next batch of suggestions, but doing it locally. I'll be adding them as a review in the next hour or two!

File: source/reference-manual/security/secure-elements/secure-element.tpm.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+To register with support for TPM 2 PKCS#11:
+
+.. code-block:: shell
+
+    lmp-device-register -n <device-name> -m /usr/lib/pkcs11/libtpm2_pkcs11.so -S <so-pin> -P <user-pin>
+
+.. _TPM2 software stack:
+   https://tpm2-software.github.io/
+
+.. _TPM 2 Software Stack (TPM2 TSS):
```
Comment (kprosise) [inline source/reference-manual/security/secure-elements/secure-element.tpm.rst:100]:
```suggestion
.. _TPM 2 Software Stack (TSS):
```

File: source/reference-manual/security/secure-elements/secure-element.tpm.rst [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+   https://github.com/tpm2-software/tpm2-tss
+
+.. _TPM 2 PKCS#11:
```
Comment (kprosise) [inline source/reference-manual/security/secure-elements/secure-element.tpm.rst:103]:
```suggestion
.. _TPM 2 Public-Key Cryptography Standards (PKCS) #11:
```

File: source/reference-manual/security/secure-elements/secure-element.tpm.rst [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+While it is possible to leverage TPM 2 via TSS directly (e.g. using ``tpm2-tss-engine`` with OpenSSL), in order
+to have a single and common interface for the software stack used in LmP, the integration leverages the
+`TPM 2 PKCS#11`_ interface instead.
+
+TPM 2 PKCS#11 Support
+---------------------
+
+The `PKCS#11 API`_ provides a standardized way for applications to interact with security tokens, such as smart
+cards, NXP SE05X and TPM 2 devices. The `TPM 2 PKCS#11`_ implementation provides a set of functions that enable applications
```
[the remark this answers is not in the document]
Comment (ricardosalveti) [reply source/reference-manual/security/secure-elements/secure-element.tpm.rst:40]: Here we should change to Trusted Platform Module instead.

**Comment by doanac** [comment]
Docs for 250bf66cdf61dd6acedadfacd9f6837fcbf74274 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1998/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 8b6cbb7f12d0c94637f31536f4064a5276b785f9 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1999/docs/artifacts/html/index.html

**Pull request merged**